interview-questions.md

```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: In addition to my previous response, it's also worth noting that both map and filter return SUBSETS of the original array, which means that the original array will not be mutated.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is repeating a process over and over, and in coding terms, iteration will usually stop when a specified condition is met. You see iteration in Javascript through the use of things like for loops, in loops, map, and filter. Iterating is an incredibly useful skill for making sense of data, because we are often dealing with VERY large amounts of it, and iteration makes searching through the sea of data to find the specific thing we're looking for much easier.
 
-Researched answer:
+Researched answer: In addition to my previous response, you can also use iteration in regards to coding to refer to the process of improving a project through the implimentation of testing and review until you're left with the end result, which will hopefully be the ideal version of what you were originally trying to achieve. (:
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: In Javascript, an object is a way of storing sets of data inside of key:value pairs. You can compare them to arrays, but instead of accessing the value inside of an object with it's index, you can access value based on the set name of its' key. Classes, on the other hand, are like empty templates that we can then fill with objects to be reusable. For example, if you wanted to store user information, you could establish a Class template with an object that included their name, email address, age, phone number, etc. Any time you want to add a new user, you'd simply declare a variable and assign it to that Class, passing in the unique information that would then be stored inside of the key:value pairs that make up the construction of the Class.
 
-Researched answer:
+Researched answer: In addition to my previous response, it's important to note that objects are also a data type in the Javascript language. They are considered non-primitive data types, along with classes.
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: That's a great question! I'm not personally familiar with the term hoisting in regards to Javascript, but based on the word "hoisting", I would infer that it has something to do with bringing something to the top of your code. I'll definitely look into it, and get back to you with a better answer!
 
-Researched answer:
+Researched answer: The process of hoisting in Javascript refers to the practice of moving all of your declarations to the top of their scope. This can be very beneficial, especially variable declaration. Although it may seem obvious, it's a very good idea to make sure that all of your variables have been declared BEFORE running your code.
```

Comment:
Crushed this section!  Your answers are really thorough and I really like how you extended in the researched answers to bring new depth to the answer.  Great work!